main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -82,9 +82,11 @@ object Namer {
        * Namespace.
        */
       case WeededAst.Declaration.Namespace(ns, uses, imports, decls, loc) =>
-        flatMapN(mergeUseEnvs(uses, ns0, uenv0, ienv0, prog0)) {
+        // Note: Opening a new namespace clears all current imports and uses.
+        // Hence we pass empty import and use environments.
+        flatMapN(mergeUseEnvs(uses, ns0, UseEnv.empty, ImportEnv.empty, prog0)) {
```

Comment:
Question: What is actually merged here if UseEvn and ImportEnv is empty? Can we omit the merge or no?

Reply:
For `use`, I think we definitely need to stick with merging, because use environments *can* be merged, for example:

```
namespace A {
    pub enum Foo
    pub enum Bar
    pub def foo(): Foo = ???
    pub def bar(x: Foo): Bar = ???
}

namespace B {
    use A.Bar
    use A.bar

    pub def baz(): Bar = {
        use A.Foo;
        use A.foo;
        let x: Foo = foo();
        bar(x)
    }
}
```

So the use environment within `baz` is a combination of the use environment within the `B` namespace, plus the use environment within the function.

And given that we're merging use environments in this case, then it makes sense to create a "fresh" use environment by merging into an empty environment as the code does above, and also [here](https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Namer.scala#L59).

For `import` the situation is different, however, because (as things currently stand, at least) use environments are now never actually merged.

But...

I still think that this is a mistake. I think that users will be confused that it's legitimate to use a function at namespace scope, but not to import a method at namespace scope. I.e.:

```
namespace A {
    use A.foo // this compiles
    import x.y.Z.foo(): Unit \ IO // this does not
```

And that it's legitimate to use a type at function scope, but not to import a type at function scope. I.e.:

```
def foo() =
    use A.Foo; // this compiles
    import x.y.Foo; // this does not
    ....
```

If we do unify the way that use and import work, then the same argument will apply to merging the UseEnv as the ImportEnv.

Reply:
> So the use environment within baz is a combination of the use environment within the B namespace, plus the use environment within the function.

OK, yes. This is the key.